Rmd/DESeq2_report_new.Rmd

```diff
@@ -203,6 +204,12 @@ for(cont in volcano_contrasts){
     passed = subset(volcano_df, contrast == cont & sig_group == volcano_status_passed) %>% mutate(log10p = -log10(padj)) %>% pull(log10p),
     RODAF = subset(volcano_df, contrast == cont & sig_group == volcano_status_RODAF) %>% mutate(log10p = -log10(padj)) %>% pull(log10p)
   )
+  volcano_labs[[cont]] <- list(
+    FDR = subset(volcano_df, contrast == cont & sig_group == volcano_status_FDR) %>% pull(Gene_Symbol),
+    FC = subset(volcano_df, contrast == cont & sig_group == volcano_status_FC) %>% pull(Gene_Symbol),
+    passed = subset(volcano_df, contrast == cont & sig_group == volcano_status_passed) %>% pull(Gene_Symbol),
+    RODAF = subset(volcano_df, contrast == cont & sig_group == volcano_status_RODAF) %>% pull(Gene_Symbol)
+  )
 }
```

Comment:
I agree with what you said earlier when suggesting this approach - there is probably a more efficient way. I think actually creating a dataframe for each contrast using some apply-like mapping function would work well (maybe even like nested data frames of some kind, I have never tried that but it might be a good use case). Here we are creating lists of the same length, which in some abstraction is basically a dataframe already...

But if it ain't broke...

Reply:
Yeah I am going to shunt all the blame for this to plotly haha. It took me a while to figure out how to construct things the way it wanted so I didn't bother trying to optimize after the fact.